pkg/log/log.go

```diff
@@ -64,6 +66,9 @@ func init() {
 		loggerWasSetLock.Lock()
 		defer loggerWasSetLock.Unlock()
 		if !loggerWasSet {
+			// Alert the user that this is not a good thing to rely on
+			fmt.Fprintln(os.Stderr, "warning: log.SetLogger was not called; this is not recommended and may be unsupported in a future version of controller-runtime")
```

Comment:
Can you mention here that they should explicitly call `SetLogger(NullLogger{})`? I would assume ppl who don't want the c-r logging will ignore that message otherwise

Reply:
+1 for actionable output.

Reply:
Good idea - the more actionable the better!  Done!